src/Metadata/class-post-builder.php

```diff
@@ -60,6 +60,7 @@ public function get_metadata(): array {
 		$this->build_publisher();
 		$this->build_keywords( $this->post );
 		$this->build_metadata_post_times( $this->post );
+		$this->build_post_ids( $this->post );
```

Comment:
Is it possible to pass in the post ID directly?

Reply:
Thanks for the suggestion! I get where you're coming from, but I think it'll be better to stick with the current pattern of passing the `WP_Post` object around. Here's why:

- All our existing methods use WP_Post, so changing to post ID would break that pattern.
- Using WP_Post keeps all the post-related data and logic neatly contained. 
- If we ever need to pass related post IDs or additional post data, we'd add more methods, increasing the risk of breaking changes.

So it sounds like maintaining the current approach of working with `WP_Post` objects makes the codebase cleaner and easier to maintain. What do you think?